ipalib/util.py

```diff
@@ -801,7 +801,16 @@ def _resolve_record(owner, rtype, nameserver_ip=None, edns0=False,
 
     res = DNSResolver()
     if nameserver_ip:
+        # When validating forwarders, nameserver_ip takes the format
+        # '<ip> port <port>', which is not a vaild IP address. In this
+        # case, split the string and add the IP part to res.nameservers,
+        # and the ip:port pair to res.nameserver_ports dict.
+        nameserver_ip, *port = tuple(
+            map(str.strip, nameserver_ip.split(" port "))
+        )
```

Comment:
This relies on the user providing the same spacing. If they include extra spacing then this will fail:
```
>>> tuple(map(str.strip, '192.168.0.1    port 53'.split(' port      ')))
('192.168.0.1    port 53',)
```
A plain split may be better, and if you want to be paranoid validate that the resulting list as (<IP>, "port", decimal < some max)

Reply:
I agree with the split problem. I fixed it by modifying multiple consecutive spaces to a single space in namesever_ip.

If invalid port is given, other validations will kick in and provide the proper errors, either before this point, or when trying to resolve the nameserver_ip.

Reply:
Nice solution